application-forum-ui/src/main/resources/ForumCode/Macros.xml

```diff
@@ -535,6 +535,15 @@ $value.replaceFirst($regextool.quote('{{html clean="false" wiki="false"}}'), '')
 ##
 ##
 #macro(getOrdredTopics $forum $start $nb $sortField $varName)
+  #set ($supportsNS = $services.extension.core.getCoreExtension('org.xwiki.platform:xwiki-platform-model').id.version.compareTo($services.extension.parseVersion('7.2')) &gt;= 0)
+  #if ($supportsNS)
+  ## define some variables to be used with Nested Spaces
+    #set($hierarchyCondition = "doc.space=:forum and ")
+    #set($hierarchyBinding = $doc.space)
```

Comment:
My concern is that before your changes, we were able to use `#getOrdredTopics()` from anywhere. Now, this macro could only be used in a forum. Imagine you want to create a gadget that displays the topics of some forum, and you put that gadget on the wiki home page. It will fail, because `$doc.space` will return `Main`, which is not a forum.

It is not very important right now for sure. I just want to know if it is possible to do the same thing without using the `$doc` variable.